src/generators/Generators.jl

```diff
@@ -27,23 +35,27 @@ export GradientBasedGenerator
 export HeuristicBasedGenerator
 export @objective, @threshold, @with_optimiser, @search_feature_space, @search_latent_space
 export JSMADescent
+export hinge_loss, invalidation_rate
+export ProbeGenerator
 
-include("functions.jl")
 include("macros.jl")
+include("utils.jl")
+
+# Optimizers
+include("optimizers/JSMADescent.jl")
 
 # Gradient-Based Generators:
 include("gradient_based/base.jl")
-include("gradient_based/functions.jl")
+include("gradient_based/utils.jl")
+include("gradient_based/loss.jl")
+
 include("gradient_based/generators.jl")
-include("gradient_based/optimisers.jl")
 include("gradient_based/probe.jl")
-export ProbeGenerator
-export hinge_loss, invalidation_rate
 
 # Non-Gradient-Based Generators:
 include("non_gradient_based/base.jl")
-include("non_gradient_based/functions.jl")
-include("non_gradient_based/generators.jl")
+
+include("non_gradient_based/feature_tweak/featuretweak.jl")
```

Comment:
`feature_tweak/featuretweak.jl` is a bit weird naming - I think the file and the folder should have names formatted in the same way.